src/widgets/hierarchical-menu/hierarchical-menu.js

```diff
@@ -46,7 +46,7 @@ hierarchicalMenu({
   [ separator=' > ' ],
   [ rootPath ],
   [ showParentLevel=true ],
-  [ limit=1000 ],
+  [ limit=30 ],
```

Comment:
It goes against the idea of having sensible defaults, which you don't have to configure. I think it's better if you have less to learn, and if you could the best results with the least effort.